src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -1364,6 +1364,25 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 		}
 	}
 
+	async updateAll(): Promise<InstallExtensionResult[]> {
+		const toUpdate: InstallExtensionInfo[] = [];
+		this.outdated.forEach((extension) => {
+			if (extension.gallery) {
+				toUpdate.push({
+					extension: extension.gallery,
+					options: {
+						operation: InstallOperation.Update,
+						installPreReleaseVersion: extension.local?.isPreReleaseVersion,
+						profileLocation: this.userDataProfileService.currentProfile.extensionsResource,
+						donotIncludePackAndDependencies: true,
+						context: { extensionIdentifier: extension.identifier }, // Provide this in result so result reporting can look it up and use its properties
```

Comment:
This is not needed because the install extension result has the extension identifier